tests/test_platform.py

```diff
@@ -35,6 +35,14 @@ def test_create_platform_error():
         _ = create_platform("nonexistent")
 
 
+def test_create_platform_from_hardware():
+    original_value = os.environ.get(PLATFORMS)
+    os.environ[PLATFORMS] = str(Path(__file__).parent)
+    _ = create_platform("dummy_hardware")
+    os.remove(Path(__file__).parent / "dummy_hardware" / PARAMETERS)
```

Comment:
Ok, the comment above was essentially to move this in a single place (a fixture), such that you can avoid duplication in the path specification, and make the mechanism atomic.

Something like:
```py
@pytest.fixture
def dummy_hardware():
    parameters = initialize_parameters(hardware=create_dummy_hardware())
    (Path(__file__).parent / PARAMETERS).write_text(
        parameters.model_dump_json(indent=4)
    )
    original_value = os.environ.get(PLATFORMS)
    os.environ[PLATFORMS] = str(Path(__file__).parent)
    yield
    (Path(__file__).parent / "dummy_hardware" / PARAMETERS).unlink()
    os.environ[PLATFORMS] = original_value
```
which would contain all the preparation code.

Reply:
Btw, for the environment variable, you do not need to handle it manually, since Pytest has a utility to do exactly that, i.e. [`MonkeyPatch.setenv`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.MonkeyPatch.setenv), turning the code above to
```py
@pytest.fixture
def dummy_hardware(monkeypatch):
    parameters = initialize_parameters(hardware=create_dummy_hardware())
    (Path(__file__).parent / PARAMETERS).write_text(
        parameters.model_dump_json(indent=4)
    )
    monkeypatch.setenv(PLATFORMS, str(Path(__file__).parent))
    yield
    (Path(__file__).parent / "dummy_hardware" / PARAMETERS).unlink()
```

Reply:
Thanks for the suggestions. Implemented in fa1b22e26d91b4d341220c3a71c5c3020d87128b.